File: python/smoke.py
"""Quick client smoke — polite delay on."""

from __future__ import annotations

import json

from sampath_api_docs import SampathApiDocsClient


def json_preview(data: object, limit: int = 200) -> str:
    try:
        return json.dumps(data, ensure_ascii=False)[:limit]
    except TypeError:
        return str(data)[:limit]


with SampathApiDocsClient(default_delay_seconds=1.0) as client:
    print("smoke", client.slug, "->", "exchange_rates")
    data = client.exchange_rates()
    preview = data[:200] if isinstance(data, str) else json_preview(data)
    print("ok", preview)

File: typescript/examples/smoke.ts
import { SampathApiDocsClient } from "../src/index.js";

/**
 * Live smoke — polite delay on. Run: npm run smoke
 * May fail if upstream is down / blocks this environment.
 */
async function main() {
  const client = new SampathApiDocsClient({ defaultDelayMs: 1000 });
  console.log("smoke", SampathApiDocsClient.slug, "->", "exchangeRates");
  const data = await client.exchangeRates();
  const preview = typeof data === "string" ? data.slice(0, 200) : JSON.stringify(data)?.slice(0, 200);
  console.log("ok", preview);
}

main().catch((err) => {
  console.error(err);
  process.exit(1);
});
